### DISCUSSION TITLE
Work with a cropped / masked area.

### DISCUSSION THREAD

**Discussion by Ziirf** [QuestPDF/QuestPDF#599] [Q&A]
Hello, I am currently trying to do some dynamic pdf generation for customer defined cards.
My problem is that currently customers can place elements out of bounds, meaning that when you generate the cards, they will look a little like this:
![image](https://github.com/QuestPDF/QuestPDF/assets/49737643/2c6b8b67-bd07-4bf5-b456-fb831fe99d3b)

That in itself is not a huge problem, the problem if they place it way out of bounds, then it will mess with the other cards.

So my question is:
Is there a way to work with a defined area, and display only that area. And if any elements go out of the boundary, it wont render?
(Desired result would be that test1 would only show "st1" as that is the only thing inbound)

```cs
public void RenderContent(IContainer container) {
	container
	.Height(Height)
	.Width(Width)
	.Layers(layer => {

                // the red background
		layer
		.PrimaryLayer()
		.Background(Color)
		.Height(Height)
		.Width(Width)
		.Unconstrained();

                // the elements (text and image)
		foreach (IRenderable item in Items) {
			layer 
			.Layer()
			.Unconstrained()
			.Element(item.RenderContent);
		}
	});
}

```

**Comment by MarcinZiabek** [comment]
I don't see any reason for the layout shift. Can you pleasae share a minimal/compilable code that reproduces the issue? I will happily investigate and offer proper guidance / fix the bug 😁

**Reply by Ziirf** [reply]
Sorry, I might have been unclear as to what my intentions were.
It isn't a bug, but rather a question as to if a feature exist, or if not, it might be a feature request.

As shown in the image below, there is 2 boxes, each would be a user generated box. where an image could extend out of the box, and therefore overlap another box. 

It would be nice if their was a option to have anything that isn't within the bound of the primary layer would be hidden.

(I am aware that the unconstrained allows it to go out of bound, but that is the only way I can mimic it be close to what I desire.)
![image](https://github.com/QuestPDF/QuestPDF/assets/49737643/4fa84ed7-f82c-4a70-9592-64937386943c)


```cs
public void RenderPage(PageDescriptor container) {

	container
	.Content()
	.Row(rows => {
		rows
		.ConstantItem(150)
		.Element(RenderLayers);

		rows
		.ConstantItem(10);

		rows
		.ConstantItem(150)
		.Element(RenderLayers);
	});
}

private void RenderLayers(IContainer container) {
	container
	.Height(150)
	.Width(150)
	.Layers(layer => {

		layer
		.PrimaryLayer()
		//.AddMask()
		.Background(Colors.Blue.Lighten1);

		// This unconstrained text should preferably cut off at the end of the blue box.
		layer
		.Layer()
		.Unconstrained() // instead of unconstrained, have a no wrap option, that will just cut
		.Text("This Text should be cropped instead of bleeding outside of the layer");

		// This image should preferably be cut off at the blue box.
		layer
		.Layer()
		.TranslateY(50)
		.TranslateX(-50)
		.Unconstrained()
		.Height(150)
		.Width(150)
		.Placeholder();
	});
}
```
